yui/src/rooms/js/rooms.js

```diff
@@ -108,14 +108,18 @@ M.mod_bigbluebuttonbn.rooms = {
                 });
             };
             btn = document.getElementById("id_password_clear");
-            btn.addEventListener('click', function(){
-                var input = document.getElementById("id_password");
-                input.value = null;
-            });
+            if (btn) {
```

Comment:
I'm not a JS expert, but just to settle my curiosity, in what scenario would the html elements not yet be loaded?

If there is a chance they will not be loaded initially, does that infer that these listeners may be set up multiple times? That could cause some unreliable behaviour.

Reply:
The HTML would not be loaded when they do not have access to change certain settings. For example you might only make certain configuration options available in the form when you have the capabilities/permissions to change them. For everyone else, they shouldn't need to see or worry about these additional fields.

Reply:
In this particular case, the clear button wouldn't be available if the site-level settings require access codes be mandatory, as clearing the access code would be unintended, whether done as part of the click handler (e.g. if ajax updated) or if it clears the field and the user clicks on update.